src/rpc/blockchain.cpp

```diff
@@ -722,7 +727,15 @@ static RPCHelpMan getblock()
             throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Block not found");
         }
 
-        block = GetBlockChecked(chainman.m_blockman, pblockindex);
+        havePruned = chainman.m_blockman.m_have_pruned;
+        if(havePruned) {
+            EnsureBlockNotPruned(chainman.m_blockman, pblockindex);
+            block = GetBlockChecked(pblockindex);
```

Comment:
Is there any value in keeping the cs_main lock here. In the very rare case where the block could be deleted after `EnsureBlockNotPruned` but before `GetBlockChecked`  without the lock, it seems fine to just error out as well. The only difference would the the error message.

This is done in https://github.com/bitcoin/bitcoin/pull/26308